### DISCUSSION TITLE
Math behind uncertainty of constraint parameters

### DISCUSSION THREAD

**Discussion by mbakker7** [lmfit/lmfit-py#969] [General]
One of the cool features of LMFit is that parameters can have a min and max value, even when doing Levenberg-Marquardt optimization. 
In the manual of older versions of LMFit, there was quite some math explaining how the uncertainty of these parameters was computed when the parameter was constraint by a min and max value. I could not find this math back in the current manual. Is that still there? And if it is not, is the old math still accessible somewhere?

**Comment by newville** [comment]
@mbakker7  See https://lmfit.github.io/lmfit-py/bounds.html for the explanation and https://github.com/lmfit/lmfit-py/blob/776e14d73028548f783288c37468b520e459050e/lmfit/parameter.py#L893 (and also the following `scale_gradient` method for the code that implements this.

**Reply by mbakker7** [reply]
Thanks, @newville , but I seem to recall from the old manual that you included an analysis with derivatives to explain how the covariance is "re-transformed" to estimate the uncertainties of the original un-transformed parameters. Do I recall that correctly, and if so, is that analysis still available somewhere?

**Comment by newville** [comment]
@mbakker7 Maybe https://github.com/lmfit/lmfit-py/blob/master/lmfit/minimizer.py#L761

in which `Parameter.scale_gradient` is doing most of the heavy lifting.  I don't we removed anything like that from the documentation.  Maybe that was in the documentation for leastsqbound, which is where we got some of the original ideas for the covariance transformation.

**Reply by mbakker7** [reply]
My memory is probably bad, but I am giving it a last try. Do you happen to have a pdf of version 0.8 or 0.9? I'd love to take a look at those. I tried to re-generate those from the repositories, but that turns out not to be so easy.

**Comment by newville** [comment]
No, I do not.  The GitHub repo will go back that far, and you are welcome to browse the older versions of the codes and documentation all you want. I can not provide any more help than that.

**Comment by mbakker7** [comment]
I really appreciate your help. Thanks for a great package.
